openvdb_points/unittest/TestAttributeSet.cc

```diff
@@ -189,6 +191,60 @@ TestAttributeSet::testAttributeSetDescriptor()
         CPPUNIT_ASSERT(descr1->hasSameAttributes(*descr4));
     }
 
+    { // Test uniqueName
+        Descriptor::Inserter names;
+        names.add("test", AttributeS::attributeType());
+        names.add("test1", AttributeI::attributeType());
+
+        Descriptor::Ptr descr1 = Descriptor::create(names.vec);
+
+        const openvdb::Name uniqueName1 = descr1->uniqueName("test");
+        CPPUNIT_ASSERT_EQUAL(uniqueName1, openvdb::Name("test0"));
+
+        const Descriptor::NameAndType newAttr(uniqueName1, AttributeI::attributeType());
+        Descriptor::Ptr descr2 = descr1->duplicateAppend(newAttr);
+
+        const openvdb::Name uniqueName2 = descr2->uniqueName("test");
+        CPPUNIT_ASSERT_EQUAL(uniqueName2, openvdb::Name("test2"));
+    }
+
+    { // TODO: Test hasGroup(), setGroup(), dropGroup(), clearGroups()
+        Descriptor descr;
+
+        CPPUNIT_ASSERT(!descr.hasGroup("test1"));
+
+        descr.setGroup("test1", 1);
+
+        CPPUNIT_ASSERT(descr.hasGroup("test1"));
+        CPPUNIT_ASSERT_EQUAL(descr.groupMap().at("test1"), size_t(1));
+
+        descr.setGroup("test5", 5);
+
+        CPPUNIT_ASSERT(descr.hasGroup("test1"));
```

Comment:
Think this was just checking test1 hadn't been affected, I've extended this test a little anyway though.